framework/src/Degradations/PhantomCharacter.cpp

```diff
@@ -987,11 +979,11 @@ phantomCharacter(const cv::Mat &imgOriginal, Frequency frequency)
 }
 
 QImage
-phantomCharacter(const QImage &imgOriginal, Frequency frequency)
+phantomCharacter(const QImage &imgOriginal, Frequency frequency, QString phantomPatternsPath)
```

Comment:
you should pass a const reference here:
```suggestion
phantomCharacter(const QImage &imgOriginal, Frequency frequency, const QString &phantomPatternsPath)
```